assemble/conf/accumulo-env.sh

```diff
@@ -95,7 +95,7 @@ JAVA_OPTS=("${JAVA_OPTS[@]}"
 
 case "$cmd" in
   monitor|gc|master|tserver|tracer)
-    JAVA_OPTS=("${JAVA_OPTS[@]}" "-Dlog4j.configurationFile=log4j2-service.properties")
+    JAVA_OPTS=("${JAVA_OPTS[@]}" "-Dlog4j.configurationFile=log4j2-service.properties" "-DHaltVMOnThreadError=true")
```

Comment:
Interesting strategy. I wonder if the way you're handling this bypasses our setting for `'-XX:OnOutOfMemoryError=kill -9 %p'` in these threads.

Reply:
I think it will in this specific case. Although, I could bypass Halt if the Error is an OutOfMemoryError. I believe that is the only error that has special handling in the JVM.

Reply:
I think LoggingRunnable avoided the question of which would win by trying to wrap the Runnable and catching the Throwables in the run method instead of relying on the uncaught handler. However, the fatal flaw with LoggingRunnable is that it wouldn't re-throw things. Your solution to put an uncaught handler on everything makes it so we can (should) delete LoggingRunnable, as it's not needed any more, but it raises questions about what wins. I know that it wouldn't necessarily make sense to re-throw from the handler... because the docs say stuff thrown from the handler are ignored... but not sure if that applies to Error types as well as Exceptions.

I'm finding myself with more questions than answers about how Java actually behaves, so it's hard to have an opinion about how things should go. I think I might want to simulate a few scenarios to test the various JVM behaviors, and use that to inform what to do.

Reply:
> know that it wouldn't necessarily make sense to re-throw from the handler... because the docs say stuff thrown from the handler are ignored... but not sure if that applies to Error types as well as Exceptions.

I'm pretty sure I have seen a StackOverflowError from a Thread that kills the Thread but not the VM. My guess is that an Error re-thrown from the handler on a Thread would do nothing, unless its the OutOfMemoryError and the OnOutOfMemoryError parameter is set.